lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb

```diff
@@ -42,26 +42,24 @@ def self.integrate(pod_bundle, targets)
           #         The build configuration.
           #
           def self.set_target_xcconfig(pod_bundle, target, config)
-            path = pod_bundle.xcconfig_relative_path(config.name)
-            group = config.project['Pods'] || config.project.new_group('Pods')
-            file_ref = group.files.find { |f| f.path == path }
+            file_ref = create_xcconfig_ref(pod_bundle, config)
+
             existing = config.base_configuration_reference
 
             set_base_configuration_reference = ->() do
-              file_ref ||= group.new_file(path)
               config.base_configuration_reference = file_ref
             end
```

Comment:
This block of code should probably be inlined now since it's one line - it's only called in 2 places below this.